### DISCUSSION TITLE
How to make API call to Khoj backend from a web page?

### DISCUSSION THREAD

**Discussion by lakshmajee** [khoj-ai/khoj#612] [General]
I have a react.js front end web application and it has its own chat interface.
I want to make API calls to khoj API.
How can I do that? will that be simple curl request with token available in khoj interface ?
```js
${hostURL}/api/chat?q=${encodeURIComponent(query)}&n=${resultsCount}&client=web&stream=true`;
```

**Comment by JOduMonT** [comment]
Hi;

on a self hosted Khoj; (*I don't think the API is open on the Cloud version*)

#### for me it worked well with a header
- Content-Type: application/json
- Authorisation: Bearer YOUR Khoj API

then I was able to..

### Chat (/api/chat)
````
curl -X POST http://localhost:42110/api/chat \
  -H "Content-Type: application/json" \
  -d '{"q": "Your query", "agent": "agent-name"}'
````

### Search (/api/search)
`curl "http://localhost:42110/api/search?q=term&file-type=org"`


### Automation (/api/automation)
````
curl -X POST http://localhost:42110/api/automation \
  -H "Content-Type: application/json" \
  -d '{"q": "Generate daily sales report", "crontime": "30 9 * * *"}'
````

Apparently it is also possible to
- Upload or sync files/folders (/api/content)
- Retrieve conversation history (/api/conversation)
- Summarize a document (/api/summarize)

But I`m still working on my discovery

**Comment by debanjum** [comment]
Hey, you can see the API swagger docs on your self-hosted setup at http://localhost:42110/docs#/ and the redoc at khoj cloud at https://app.khoj.dev/redoc

**Reply by JOduMonT** [reply]
thanks for the redoc
for the local swagger, I don't have access; I think it's because i use the khoj-cloud version

**Reply by debanjum** [reply]
Yeah, cloud version just has redoc. Swagger docs are turned off for production
